### ISSUE TITLE
Possible to run watcher process from Bitbar (forking, zombies, etc.)?

### ISSUE THREAD

**Issue by ZBiener** [matryer/xbar#523]
Hi All,
I'd like to start a watcher process (fswatch) from a Bitbar shell script. However, because the process doesn't exit, Bitbar can't handle the remainder of the script properly. I've tried forking a variety of forking tricks (using &, I'm not an expert), but  this doesn't seem to help, Bitbar always waits for the process to finish. Any ideas?

Here is the script. The initial echo commands don't render properly, as long as the fswatch process is running. I'm omitting some variables, for brevity's sake. Thanks!

```
echo "♺" 
echo "---"
echo "Stop Processing| size=14 bash='$script' param1=pause terminal=false"

export PATH=$HOME/bin:/usr/local/bin:/Library/TeX/texbin:$PATH
WATCH_DIR="/Users/xxx/Temp/LaTex_Compile"
LATEXCMD="/Users/xxx/Library/Scripts/synctexmk.sh"

#Kill any existing watch process, or bitbar will continue spawning them whenever the script loads.
ps aux | grep "fswatch" | grep "tex" | awk '{print $2}' | xargs kill

CMD="fswatch -0 --event Updated $WATCH_DIR/*.tex | xargs -0 -n 1 -I {} $LATEXCMD {}" 

eval $CMD
```

**Comment by gingerbeardman** [comment]
> I've tried forking a variety of forking tricks (using &, I'm not an expert)

Can you list what you have tried and the outcome to help us help you?

**Comment by ZBiener** [comment]
Yes, sorry, I should have thought of that. I'm sure these look absurd to someone who knows, but I've tried:
```
eval $CMD &
( eval $CMD ) & 
(( eval $CMD ) & ) &
```
I've tried these with NOHUP variants, removing eval and substituting the full command, and using &! and & disown instead of a single &. All of these work find on the command line -- they return to the shell without waiting for the process to finish. But it doesn't work when called from Bitbar. THANKS.

**Comment by gingerbeardman** [comment]
Thanks.

So you want to display output as it appears, as part of the output stream? Rather than wait for all the output and then display it? Hmm, not sure it is possible using BitBar but I will continue to think about it.

What is your actual goal? To notify when the LaTeX compile process is complete? Or to have some visual of how far through the compile the process is? What does the command output look like?

**Comment by ZBiener** [comment]
Thank you. I was hoping for something simpler: let the fswatch process do its work in the background, no need to capture the output. Latexmk outputs to a log file anyway. All I want from Bitbar is an indication that the watch process is running, and then the option to stop it. I didn't build in the logic for start/stopping yet, since it any echo command output was delayed anyway. [I suppose I can build fancier options later, but no need now].

So, basically, just a menu w/ a start/stop command. Thanks again.

**Comment by soncaokim** [comment]
I’m having a similar need, i.e. having a long running backup process (minutes to hours to complete), that is started at programmed time or manually.

I’ve implemented separate scheduling system, bitbar is only used to repeatedly peek at the running status and show the status/log/exit status in the menu as feed back.

Unfortunately it is not in a state that I can share the code (there are private information mixed in the scripts), the basic idea is to have a Bitbar script doing the following:

• If the process is not started at programmed time, it will be started and detached from the script.
• The PID and output are captured in a file.
• If the process is still running, the Bitbar script looks at the output log and format it to display in a submenu.
• In addition to the output log, it uses the PID and the eventual exit status to see if the backup process has terminated or still running and decorate it accordingly in the bibar menu.

The Bitbar script is hence short-lived and is started by Bitbar every few seconds.

**Comment by tevino** [comment]
Here is what I did to launch a daemon in a plugin

https://gist.github.com/tevino/3ae37b0aa1f70979422f928186220870
